extensions/amp-lightbox/0.1/amp-lightbox.js

```diff
@@ -150,9 +178,90 @@ class AmpLightbox extends AMP.BaseElement {
     this.active_ = false;
   }
 
+  /**
+   * Handles scroll on the amp-lightbox.
+   * @private
+   */
+  scrollHandler_() {
+    const currentScrollTop = this.element./*OK*/scrollTop;
+    this.pos_ = currentScrollTop;
+
+    if (this.scrollTimerId_ === null) {
+      this.waitForScroll_(currentScrollTop);
+    }
+  }
+
+  /**
+   * @param {!number} startingScrollTop
+   * @private
+   */
+  waitForScroll_(startingScrollTop) {
+    this.scrollTimerId_ = timerFor(this.win).delay(() => {
+      if (Math.abs(startingScrollTop - this.pos_) < 30) {
+        dev().fine(TAG, 'slow scrolling: ' + startingScrollTop + ' - '
+            + this.pos_);
+        this.scrollTimerId_ = null;
+        this.update_(this.pos_);
+      } else {
+        dev().fine(TAG, 'fast scrolling: ' + startingScrollTop + ' - '
+            + this.pos_);
+        this.waitForScroll_(this.pos_);
+      }
+    }, 100);
+  }
+
+  /**
+   * @param {number} pos
+   * @param {function(!Element)} callback
+   * @private
+   */
+  forEachInLightbox_(pos, callback) {
+    const containerHeight = this.element./*OK*/clientHeight;
+    for (let i = 0; i < this.children_.length; i++) {
+      const cell = this.children_[i];
+      if (cell./*OK*/offsetTop + cell./*OK*/offsetHeight >= pos &&
+          cell./*OK*/offsetTop <= pos + containerHeight) {
+        callback(cell);
+      }
+    }
+  }
+
+  /**
+   * @param {number} newPos
+   * @param {number} oldPos
+   * @private
+   */
+  updateChildrenInViewport_(newPos, oldPos) {
+    const seen = [];
+    this.forEachInLightbox_(newPos, cell => {
+      seen.push(cell);
+      this.updateInViewport(cell, true);
+    });
+    if (oldPos != newPos) {
+      this.forEachInLightbox_(oldPos, cell => {
+        if (seen.indexOf(cell) == -1) {
+          this.updateInViewport(cell, false);
+          this.schedulePause(cell);
```

Comment:
I don't think we want to do this. It may make sense for scrollable-carousel but for a scrollable container, I don't think so given runtime does not `pause` elements when they go out of viewport.